web/api/backend_test.go

```diff
@@ -25,7 +26,35 @@ func TestHandlerAuthenticateCredentialsTokenValid(t *testing.T) {
     "token":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJVcGxvYWRDcmVkZW50aWFscyI6eyJxdW90YV9wcm9qZWN0X2lkIjoiZGVsYXlzLW9yLXRyYWZmaS0xNTY5MTMxMTUzNzA0IiwicmVmcmVzaF90b2tlbiI6IjEvLzBkRlN4eGk0Tk9UbDJDZ1lJQVJBQUdBMFNOd0YtTDlJcmE1WVRubkZlcjFHQ1pCVG9Ha3dtVk1Bb3VuR2FpX3g0Q2dId01BRmdGTkJzUFNLNWhCd3hmcEduODh1M3JvUHJSY1EiLCJ0eXBlIjoiYXV0aG9yaXplZF91c2VyIn0sImV4cCI6OTIyMzM3MTk3NDcxOTE3OTAwN30.qV78DszW3-c9g3LdT8uxOkgeufHg51QUhB2FjG2HmBY"
 	}`)
 
-	req, err := http.NewRequest("GET", "/", bytes.NewBuffer(jsonStr))
+	req, err := http.NewRequest("POST", "/", bytes.NewBuffer(jsonStr))
+	if err != nil {
+		t.Fatal(err)
+	}
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(Handler)
+
+	// Our handlers satisfy http.Handler, so we can call their ServeHTTP method
+	// directly and pass in our Request and ResponseRecorder.
+	handler.ServeHTTP(rr, req)
+
+	if status := rr.Code; status != http.StatusOK {
+		t.Errorf("handler returned wrong status code: got %v want %v",
+			status, http.StatusOK)
+	}
+}
+
+func TestHandlerAuthenticateCredentialsTokenValidNoCredentials(t *testing.T) {
+	jsonStr := []byte(`{
+        "commandtype":"fetch",
+        "args":{
+            "scope":["cloud-platform","userinfo.email"]
+		},
+    "cachetoken": false,
```

Comment:
Is `cachetoken` still a field we want to check against? I think you changed it to credential. Also, we can change this later, but we wont be passing the token with the cached credential in the body, it should be in the header, `Authorization: Bearer + token`

Reply:
We still need cache token to let it be known that a token should not be created.

Reply:
cacheToken is still needed to indicate that a token should not be created